MinimalMVVM/ViewModels/Presenter.cs

```diff
@@ -1,47 +1,116 @@
+     *      - ConvertTextCommand.Execute() ends up pointing to Presenter.ConvertText()
+     *          - When Presenter.ConvertText() is called, it shoves Presenter.SomeText through Presenter._textConverter.ConvertText(), which returns the converted all-caps string
+     *          - Then Presenter.ConvertText() adds the converted string to Presenter._history, which is data bound (via the Presenter.History property) to the history ListBox in the XAML UI
+    */ 
+
+    // The contents of Presenter are best understood when read from top to bottom!
+    // It's easy to miss, but the actual instance of Presenter that we use is created in XAML, not code-behind, in ConvertWindow.xaml
+    public class Presenter : ObservableObject // Base on the ObservableObject class so simple properties will play nice with data binding (see ObservableObject.cs)
     {
-        private readonly TextConverter _textConverter = new TextConverter(s => s.ToUpper());
-        private string _someText;
-        private readonly ObservableCollection<string> _history = new ObservableCollection<string>();
+        #region Private TextConverter Instance
+        // Create an instance of TextConverter, using an anonymous method as a parameter to pass in the conversion function since it's so simple
+        private readonly TextConverter _textConverter = new TextConverter(
+            delegate(string s)
+            {
+                // Uncomment to see when this runs
+                //Console.WriteLine("Anonymous method within TextConverter instance ran!");
+
+                return s.ToUpper();
+            }
+        );
+        // ^^^This could be made way shorter using lambda expressions as below, but for the sake of clarity I have included the more verbose, non-lambda way. Both ways give the same result.
+        // private readonly TextConverter _textConverter = new TextConverter(s => s.ToUpper());
+        #endregion
 
+        #region SomeText (User Input) Property
+        // SomeText is the property the input TextBox is data-bound to
+        // _someText is just the backend variable for SomeText
+        private string _someText;
         public string SomeText
         {
             get { return _someText; }
+
+            // This is called (because of the binding) each time the user changes the contents of the input TextBox
             set
             {
+                // Set backend variable
                 _someText = value;
+                // Raise the property changed event for the property "SomeText"
                 RaisePropertyChangedEvent("SomeText");
             }
         }
+        #endregion
+
+        #region History ObservableCollection
 
+        // Create a backend ObservableCollection variable, which is basically a WPF Data-binding-friendly List
+        // ObservableCollections implement the INotifyPropertyChanged interface so they play nice with bindings
```

Comment:
They also implement `INotifyCollectionChanged` to tell things that items in the collection have been added or removed.